server/middleware/checkPrisonerInCaseloadMiddleware.ts

```diff
@@ -0,0 +1,91 @@
+import createError from 'http-errors'
+import { RequestHandler } from 'express'
+import { DpsRoles } from './roleBasedAccessControl'
+import { userHasRoles } from '../utils/userRoles'
+import { isActiveCaseLoad, isInUsersCaseLoad } from '../utils/userCaseLoads'
+
```

Comment:
This middleware is heavily based on the [HMPPS Prisoner Profile](https://github.com/ministryofjustice/hmpps-prisoner-profile/blob/main/server/middleware/checkPrisonerInCaseloadMiddleware.ts) version, with the changes being based around the use of our types and imports etc.

Personally I'm not sure I would have written it this way, eg: with nested functions etc, but it is what it is, and the fact that I have gone through this line by line to work out what it does in order to write the tests, is probably a good thing.

I guess on the plus side, if we do decide to refactor this, we have the unit tests above to protect us 👍  - what ever changes we make, the tests should still pass 👍